src/lib/plugin/types/lens.ts

```diff
@@ -133,7 +134,7 @@ export type RuntimeContributions<C extends {} = any> = {
   }
 }
 
-export type TesttimeContributions = Utils.DeepPartial<Testing.TestContextCore>
+export type TesttimeContributions = DeepPartial<Testing.TestContextCore> & { [prop: string]: any }
```

Comment:
What does the `&` index type achieve here? I see it was in our previous implementation of `DeepPartial`. But can't remember still what it was for, and why the new version chose to not have that feature.

Reply:
Without it, plugins wouldn't be able to augment the `TestContextCore` type. They would only be able to provide different values for the existing type.
This makes sure `nexus-plugin-prisma` for example, can augment the type with a `db` property that is not on the `TestContextCore`.

The reason the `utility-types` version doesn't implement it is because it has nothing to do with `DeepPartial`.

The net result of `DeepPartial` + `& { [x: string]: any }` is:
- `DeepPartial`: Ability to get autocompletion for the existing fields of TestContextCore
- `& { [x: string]: any }`: Ability to add new arbitrary fields